include/mockturtle/views/cost_view.hpp

```diff
@@ -0,0 +1,318 @@
+/* mockturtle: C++ logic network library
+ * Copyright (C) 2018-2021  EPFL
+ *
+ * Permission is hereby granted, free of charge, to any person
+ * obtaining a copy of this software and associated documentation
+ * files (the "Software"), to deal in the Software without
+ * restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following
+ * conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ * OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/*!
+  \file cost_view.hpp
+  \brief Implements various cost estimation methods for a network
+
+  \author Hanyu Wang
+*/
+
+#pragma once
+
+#include "../networks/events.hpp"
+#include "../traits.hpp"
+#include "../utils/node_map.hpp"
+#include "../utils/recursive_cost_functions.hpp"
+#include "immutable_view.hpp"
+
+#include <cstdint>
+#include <vector>
+
+namespace mockturtle
+{
+
+struct cost_view_params
+{
+  /*! \brief Take complemented edges into account for depth computation. */
+  bool count_complements{ false };
+};
+
+/*! \brief Implements `get_cost` methods for networks.
+ *
+ * This view computes the cost of the entire network, a subnetwork, and 
+ * also fanin cone of a single node. It maintains the context of each 
+ * node, which is the aggregated variables that affect the cost a node. 
+ * 
+ * The `get_cost` method has 3 different usages:
+ * - `get_cost()` returns the cost of the entire network
+ * - `get_cost( node n )` returns the cost of the fanin cone of node `n`
+ * - `get_cost( node n, std::vector<signal> leaves )` returns the cost
+ * of a subnetwork from `leaves` to node `n`
```

Comment:
Why don't you document the individual functions directly in the normal way?
Also, the other public member functions should also be documented.